File: backend/internal/model/subscription.go
package model

import "time"

// ParseFormat 解析格式类型
type ParseFormat string

const (
	ParseFormatAuto      ParseFormat = "auto"      // 自动检测
	ParseFormatBase64    ParseFormat = "base64"    // Base64编码节点列表
	ParseFormatClash     ParseFormat = "clash"     // Clash配置文件
	ParseFormatSurge     ParseFormat = "surge"     // Surge配置文件
	ParseFormatQuantumult ParseFormat = "quantumult" // Quantumult格式
	ParseFormatSSD       ParseFormat = "ssd"       // SSD格式
)

// Subscription 订阅模型
type Subscription struct {
	ID             string      `json:"id"`
	Name           string      `json:"name"`
	URL            string      `json:"url"`
	Type           string      `json:"type"`
	ParseFormat    ParseFormat `json:"parseFormat"`    // 解析格式
	Status         string      `json:"status"`
	LastUpdate     time.Time   `json:"lastUpdate"`
	NodeCount      int         `json:"nodeCount"`
	AutoUpdate     bool        `json:"autoUpdate"`
	UpdateInterval int         `json:"updateInterval"` // 更新间隔（小时）
	CreatedAt      time.Time   `json:"createdAt"`
	UpdatedAt      time.Time   `json:"updatedAt"`
}

// SubscriptionCreateRequest 创建订阅请求
type SubscriptionCreateRequest struct {
	Name           string      `json:"name" binding:"required"`
	URL            string      `json:"url" binding:"required"`
	Type           string      `json:"type"`
	ParseFormat    ParseFormat `json:"parseFormat"`    // 解析格式
	AutoUpdate     bool        `json:"autoUpdate"`
	UpdateInterval int         `json:"updateInterval"` // 默认24小时
}

// SubscriptionUpdateRequest 更新订阅请求
type SubscriptionUpdateRequest struct {
	Name           string      `json:"name"`
	URL            string      `json:"url"`
	Type           string      `json:"type"`
	ParseFormat    ParseFormat `json:"parseFormat"`    // 解析格式
	AutoUpdate     bool        `json:"autoUpdate"`
	UpdateInterval int         `json:"updateInterval"`
}

File: frontend/src/types/subscription.ts
/**
 * 订阅类型定义
 */
export interface Subscription {
  id: string
  name: string
  url: string
  type: SubscriptionType
  parseFormat: ParseFormat
  autoUpdate: boolean
  updateInterval: number // 更新间隔（小时）
  lastUpdate: string
  nodeCount: number
  status: SubscriptionStatus
  createdAt: string
  updatedAt: string
}

export type SubscriptionType = 'ss' | 'ssr' | 'vmess' | 'vless' | 'trojan' | 'hysteria' | 'mixed'

export type SubscriptionStatus = 'active' | 'inactive' | 'updating' | 'error'

/**
 * 解析格式类型
 */
export type ParseFormat = 'auto' | 'base64' | 'clash' | 'surge' | 'quantumult' | 'ssd'

/**
 * 解析格式选项
 */
export const PARSE_FORMAT_OPTIONS: { value: ParseFormat; label: string; description: string }[] = [
  { value: 'auto', label: '自动检测', description: '自动识别订阅格式，推荐使用' },
  { value: 'base64', label: 'Base64', description: 'Base64编码的节点链接列表（v2rayN格式）' },
  { value: 'clash', label: 'Clash', description: 'Clash/Mihomo配置文件格式（YAML）' },
  { value: 'surge', label: 'Surge', description: 'Surge配置文件格式' },
  { value: 'quantumult', label: 'Quantumult', description: 'Quantumult/Quantumult X格式' },
  { value: 'ssd', label: 'SSD', description: 'SSD订阅格式（ShadowsocksD）' },
]

/**
 * 创建订阅请求
 */
export interface CreateSubscriptionRequest {
  name: string
  url: string
  type: SubscriptionType
  parseFormat?: ParseFormat
  autoUpdate?: boolean
  updateInterval?: number
}

/**
 * 更新订阅请求
 */
export interface UpdateSubscriptionRequest {
  name?: string
  url?: string
  type?: SubscriptionType
  parseFormat?: ParseFormat
  autoUpdate?: boolean
  updateInterval?: number
}
